File: lei_lds2122_grupo2-main/Frontend/src/app/viewModels/Types/RouteResumeType.ts
import { RoutePlaceType } from "./RoutePlaceType";

/**
 * Represents the route's Resume
 */
export class RouteResumeType {
    
    /**
     * Construct's an instance of {@link RouteResumeType}.
     * 
     * @param name route's name
     * @param routeID route's id
     * @param averageRating route's average rating
     * @param userName route's creator user name
     * @param profilePhoto route's creator profile photo url
     * @param isFavourite specifies if the route is marked as favorite by the user.
     * @param wasUsed specifies if the route was alrealy used by the user
     * @param description route's description
     * @param numberOfUses route's number of uses
     * @param userEvaluation user evaluation of the route
     * @param places route's places
     */
    constructor(
        public name: string,
        public routeID: number,
        public averageRating: number,
        public userName: string,
        public isFavourite: boolean,
        public wasUsed: boolean,
        public description: string,
        public numberOfUses: number,
        public userEvaluation: number,
        public places: Array<RoutePlaceType>,
        public profilePhoto?: string
    ) { }
}   

File: lei_lds2122_grupo2-main/Backend/Data/ViewModels/Types/RouteResumeType.cs
using Backend.Data.ViewModels.Types;
namespace Backend.Data.ViewModels.Types
{
    /// <summary>
    /// Struct that repesents the resume of a route 
    /// </summary>
    public class RouteResumeType
    {
        public string Name { get; set; }
        public long RouteID { get; set; }
        public double AverageRating { get; set; }
        public string UserName { get; set; }
        public string ProfilePhoto { get; set; }
        public Boolean IsFavourite { get; set; }
        public Boolean WasUsed { get; set; }
        public string Description { get; set; }
        public int NumberOfUses { get; set; }
        public long UserEvaluation { get; set; }
        public List<RoutePlaceType> Places { get; set; }


        /// <summary>
        /// Class' constructor method
        /// </summary>
        /// <param name="name">Route's name</param>
        /// <param name="userName">User name of route's creator</param>
        /// <param name="profilePhoto">Profile photoe of route's creator</param>
        /// <param name="isFavourite">Whether or not the route is a favourite route of the user that is authenticated</param>
        /// <param name="wasUsed">Whether or not the route was used by the user that is authenticated</param>
        /// <param name="userEvaluation">The evaluation of the user that is authenticated or -1 if he didn't evaluate the route</param>
        /// <param name="routeResume">Route's resume</param>
        /// <param name="routeID">Route's id</param>
        public RouteResumeType(string name, string userName, string profilePhoto, bool isFavourite,
        bool wasUsed, int userEvaluation, long routeID, double averageRating, string description, int numberOfUses, List<RoutePlaceType> places)
        {
            this.Name = name;
            this.UserName = userName;
            this.ProfilePhoto = profilePhoto;
            this.IsFavourite = isFavourite;
            this.WasUsed = wasUsed;
            this.RouteID = routeID;
            this.AverageRating = averageRating;
            this.Description = description;
            this.NumberOfUses = numberOfUses;
            this.UserEvaluation = userEvaluation;
            this.Places = places;
        }

        public RouteResumeType()
        {
            this.Name = null!;
            this.UserName = null!;
            this.ProfilePhoto = null!;
            this.Description = null!;
            this.Places = null!;
        }

    }
}
